### PR TITLE
[IMP] util/pg: remove constraint without table prefix

### PR DESCRIPTION
**Pull request by aj-fuentes**
Constraint by default have the table name as prefix. It's a common error to forget that.

When a table is renamed it's expected that ir_model_constraint may be missing. The ORM may not know about it yet.

See odoo/upgrade#5915

### CODE REVIEW

File: src/util/pg.py [hunk 1]
```diff
@@ -608,18 +608,45 @@ def view_exists(cr, view):
...
-        cr.execute("DELETE FROM ir_model_data WHERE model = 'ir.model.constraint' AND res_id IN %s", [ids])
+    if not cr.rowcount:
+        if name.startswith(table + "_"):
+            log("%r not found in ir_model_constraint, table=%r", name, table)
+            return False
+        log("%r not found in ir_model_constraint, attempting to remove with table %r prefix", name, table)
+        return remove_constraint(cr, table, "{}_{}".format(table, name), cascade, warn)
+    ids = tuple(c for (c,) in cr.fetchall())
+    cr.execute("DELETE FROM ir_model_data WHERE model = 'ir.model.constraint' AND res_id IN %s", [ids])
+    return True
```
Comment (Pirols) [inline src/util/pg.py:649]:
```suggestion
    if cr.rowcount:
        ids = tuple(c for (c,) in cr.fetchall())
        cr.execute("DELETE FROM ir_model_data WHERE model = 'ir.model.constraint' AND res_id IN %s", [ids])
        return True
    if name.startswith(table + "_"):
        log("%r not found in ir_model_constraint, table=%r", name, table)
        return False
    log("%r not found in ir_model_constraint, attempting to remove with table %r prefix", name, table)
    return remove_constraint(cr, table, "{}_{}".format(table, name), cascade, warn)
```

Comment (Pirols) [reply src/util/pg.py:649]: Arguments:

* Avoid changing the blame for the normal case
* No nested `if`s
* `fetchall()` right after query

File: src/util/pg.py [hunk 2]
```diff
@@ -608,18 +608,45 @@ def view_exists(cr, view):
...
+
+    This function removes the constraint `name` from the table named `table`. It also
+    removes records from `ir_model_constraint` and its xmlids. It logs not found
+    constraints.
+
+    .. note::
+
+       If there is no constraint `name`, this function will attempt to remove
+       ``{table}_{name}``, the latter is the default name the ORM uses for constraints
+       created from `_sql_constraints`. This name mismatch will be logged.
```
Comment (Pirols) [inline src/util/pg.py:623]:
```suggestion
       If there is no constraint `name` on `table`, this function will attempt to remove
       ``{table}_{name}``, the latter is the default name the ORM uses for constraints
       created from `_sql_constraints`.
```

I wouldn't add the last ambiguous part to this note. In fact, mismatches are logged iff `name` is not found in the first place.

**Review by Pirols — COMMENTED** [review]
IMO that's a good change. I commented about a couple of detais

File: src/util/pg.py (hunk 1 above)
Comment (aj-fuentes) [reply src/util/pg.py:649]: Indeed, this code is a remnant of a non-recursive version I had first...

File: src/util/pg.py (hunk 2 above)
Comment (aj-fuentes) [reply src/util/pg.py:623]: I removed it, although it was _not_ ambiguous: name mismatches are always logged, in the form "... not found in ..." ;)

**Review by Pirols — APPROVED** [review]
lgtm

**Review by KangOl — APPROVED** [review]
@robodoo r+

**Comment by KangOl** [comment]
Hey!
But runbot is red :(

@robodoo r-

**Comment by KangOl** [comment]
upgradeci retry with always only account account_disallowed_expenses account_edi_proxy_client account_intrastat appointment base calendar document documents documents_spreadsheet fleet hr hr_holidays hr_timesheet hr_work_entry l10n_lu_reports l10n_rs mail onboarding project_forecast repair sale sale_project sale_purchase sale_renting sale_subscription sale_timesheet spreadsheet_edition stock survey website_slides website_slides_survey

**Comment by aj-fuentes** [comment]
upgradeci retry

**Comment by aj-fuentes** [comment]
upgradeci retry

**Comment by aj-fuentes** [comment]
upgradeci retry

**Comment by aj-fuentes** [comment]
See https://github.com/odoo/upgrade/pull/5915

**Comment by KangOl** [comment]
@robodoo r+

**Pull request closed without merging**